scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/EditProjectController.java

```diff
@@ -0,0 +1,51 @@
+package edu.asu.diging.scriptoocloud.web;
+
+import java.security.Principal;
+import java.util.Map;
+import java.util.Optional;
+
+import javax.validation.Valid;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+import edu.asu.diging.scriptoocloud.core.data.ProjectRepository;
+import edu.asu.diging.scriptoocloud.core.model.Project;
+import edu.asu.diging.scriptoocloud.core.model.impl.ProjectImpl;
+import edu.asu.diging.scriptoocloud.core.service.ProjectManager;
+
+@Controller
+public class EditProjectController {
+
+    @Autowired
+    private ProjectManager projectManager;
+    
+    @RequestMapping(value = "/auth/edit/{id}", method = RequestMethod.GET)
+    public String get(@Valid @ModelAttribute("project") Optional<ProjectImpl> projectImpl, BindingResult result,
+            Model model, RedirectAttributes redirectAttrs, Principal principal, @PathVariable("id") int id) {
+        if (result.hasErrors()) {
+            model.addAttribute("project", new ProjectImpl());
```

Comment:
you want to pass `projectImpl` here or the user will loose all the changes they already made

Reply:
actually, there shouldn't be any errors since at this point nothing has been submitted yet.

Reply:
Ok. Are you saying I should remove this if statement then?

Reply:
yes, the get method does not need to validate the project.